eth_client/lib/eth_client/abi.ex

```diff
@@ -2,10 +2,57 @@ defmodule EthClient.ABI do
   @moduledoc """
   """
   alias EthClient.Context
+  alias EthClient.Rpc
+
+  # get whether it's an etherscan-linked bc: get_etherscan_api_key...
+  def get("0x" <> _ = address) do
+    if Context.etherscan_api_key() do
+      get_etherscan(address)
+    else
+      get_non_etherscan(address)
+    end
+  end
 
-  def get("0x" <> _ = address), do: get_etherscan(address)
   def get(abi_path), do: get_local(abi_path)
 
+  def to_selector(function_def) do
+    selector = ABI.FunctionSelector.decode(function_def["name"])
```

Comment:
you could  "ABI.FunctionSelector.decode(function_def["name"])" directly on a pipe like this :

`
      ABI.FunctionSelector.decode(function_def["name"])
      |> Map.put(:method_id, function_def["selector"])
      |> Map.put(:state_mutability, function_def["stateMutability"])

`
thus avoiding the use of "selector" variable